File: challenges/codeeval/medium/046/victor.ayalamarin.java

package codeevalmedium;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
/**
 *
 * @author VICAYALA82
 */
public class PrimosLess {
    public static void main(String[] args) throws FileNotFoundException {
        int linea;
        String ruta="C:\\Users\\VICTOR\\Documents\\Aprueba.txt";
        File file = new File(ruta);
        //File file = new File(args[0]);
        try (Scanner leeArc = new Scanner(file)) {
           while (leeArc.hasNext()){
                int numero=5;
                int numPriv[];
                String resp="2,3";
                linea=Integer.parseInt(leeArc.nextLine().trim());
                
                if(linea==2){
                        resp="2";
                }else if(linea==3){
                        resp="2,3";
                }else{
                    numPriv=new int[2];
                    numPriv[0]=2;
                    numPriv[1]=3;
                    while (numero<linea){

                        if (linea>3){
                            
                            double raiz=Math.sqrt(numero)+1;
                            boolean ban=true;
                            int i=0;
                            boolean pri=true;
                            boolean zer=false;
                            do{
                                if(i<numPriv.length & (numero%numPriv[i]==0)){
                                    ban=false;
                                    pri=false;
                                }
                                else{
                                        i=i+1;
                                }
                            }while ((numPriv[i-1]<=raiz & ban==true)& i<numPriv.length);
                            if(pri){
                                int vecAux[]= new int[numPriv.length];
                                vecAux=numPriv.clone();
                                numPriv= new int[vecAux.length+1];
                                for(int k=0; k<vecAux.length;k++){
                                    numPriv[k]=vecAux[k];
                                }
                                numPriv[numPriv.length-1]=numero;
                                resp=resp+","+numero;
                            }
                        }
                        numero=numero+2;
                    }
                    //System.out.println(resp);
                }
                System.out.println(resp);
           }
           leeArc.close();
        }
    }
}

File: challenges/codeeval/medium/034/victor.ayalamarin.py
__author__ = "VICAYALA82"
__date__ = "$Dec 4, 2015 8:45:57 PM$"

#file = open(sys.argv[1], 'r')
file = open('C:\\Users\\VICTOR\\Documents\\Aprueba.txt','r')
for line in file:
    linea=line.strip(' \n')
    if linea:
        n=linea.find(';')
        resultado=''
        part1 = linea[:n].strip(' ').split(',')
        part2 =linea[n+1:].strip(' ')
        auxn =int(part2)
        for k in range (len(part1)):
            aux=int(part1[k])
            part1[k]=aux
        ban=False;
        for i in range(len(part1)):
            for j in range(i+1,len(part1)):
                if part1[i]+part1[j]==auxn:
                    resultado=resultado+str(part1[i])+','+str(part1[j])+';'
                    ban=True
        if ban:
            print (resultado.strip(';'))
        else:
            print ('NULL')
file.close()
